onnxruntime/lora/adapter_format/adapter_schema.fbs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace onnxruntime.adapters;
+
+// Tensor
+enum TensorDataType : int32 {
+  UNDEFINED = 0,
+  FLOAT = 1,
+  UINT8 = 2,
+  INT8 = 3,
+  UINT16 = 4,
+  INT16 = 5,
+  INT32 = 6,
+  INT64 = 7,
+  STRING = 8,
+  BOOL = 9,
+  FLOAT16 = 10,
+  DOUBLE = 11,
+  UINT32 = 12,
+  UINT64 = 13,
+  COMPLEX64 = 14,
+  COMPLEX128 = 15,
+  BFLOAT16 = 16,
+  FLOAT8E4M3FN = 17,
+  FLOAT8E4M3FNUZ = 18,
+  FLOAT8E5M2 = 19,
+  FLOAT8E5M2FNUZ = 20,
+}
+
+// For simplicity, we will have only have one data field
+// - raw_data for all primitive types.
+// We do not foresee strings as parameters.
+table Parameter {
```

Comment:
You could consider reusing `Tensor` from ort.fbs: https://github.com/microsoft/onnxruntime/blob/d4692835bf7affb51afa83b47159f43e7526ac9d/onnxruntime/core/flatbuffers/schema/ort.fbs#L150-L166

There already are many utilities around ort tensor loading and saving which you could take advantage of.

Since this data type is opaque to the user, the user doesn't need to know what is used under the hood (whether it is a Parameter or a Tensor).

Reply:
See the comment above.